ndc-models/Cargo.toml

```diff
@@ -4,6 +4,10 @@ description = "Protocol definitions for the Hasura NDC specification"
 version.workspace = true
 edition.workspace = true
 
+[features]
+# Use Arc to nest Relations rather than Box
+arc-relation = []
```

Comment:
Is there a reason we can't pick one or the other?

Reply:
No connector will want to use `Arc`s for this. `Box` is appropriate. However, due to how these types are used in the engine, it requires `Arc` to avoid constant cloning due to the way DataFusion works. So this is a hack to allow engine to get Arcs but everybody else gets what this should be: a `Box`.

The alternative which I explored was have engine use `Arc<Mutex<..>>` so that it could avoid cloning (by using controlled mutation to move rather than clone), but that made the engine code gross. This seems like the least worst option that didn't allow weird engine concerns to pollute connectors.